collect_app/src/main/java/org/odk/collect/android/dao/FormsDao.java

```diff
@@ -171,52 +171,48 @@ public int updateForm(ContentValues values, String where, String[] whereArgs) {
 
     public List<Form> getFormsFromCursor(Cursor cursor) {
         List<Form> forms = new ArrayList<>();
-        if (cursor != null) {
```

Comment:
Can you please describe why you've made this change? They look mostly equivalent except that in the new version the cursor is not closed.

Reply:
It's because the cursor is somehow reused (no created again) if we go back from a survey (for example after saving) so I can't close that cursor because then I'm not able to read rows again.

Btw I need to investigate why we are moved back to forms list after saving a form it was always the main menu as I remember...

Reply:
Got it, thanks! That kind of info is perfect for a commit message description.

Reply:
😭 We did actually start having a conversation about this change, we just didn't go deep enough. (See https://github.com/opendatakit/collect/pull/2687#issuecomment-433176257 for later implications)